### PR TITLE
docs: modify epochs ADR to capture latest design

### PR DESCRIPTION
**Pull request by insumity**
## Description
Modified the epochs ADR to describe the latest epoch-implementation we decided upon (see [#1516](https://github.com/cosmos/interchain-security/issues/1516)). 

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct `docs:` prefix in the PR title
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary 
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)
- [ ] Checked that the documentation website can be built and deployed successfully (run `make build-docs`)

### CODE REVIEW

File: docs/docs/adrs/adr-014-epochs.md [hunk 1]
```diff
@@ -22,37 +23,45 @@ As a matter of fact, this already happens due to relaying delays.
...
-  Currently, this method is called in every block before queueing a `VSCPacket`. 
-  Also, the method uses the `KeyAssignmentReplacement` state, which is pruned at the end of every block. 
-  This needs to be done once per epoch instead.
-- At the end of every epoch, if there were validator set changes on the provider, then for every consumer chain, construct a `VSCPacket` with all the accumulated validator changes and add it to the list of `PendingVSCPackets`.
-
-As an optional change, to better accommodate [the Partial Set Security design](https://informalsystems.notion.site/Partial-Set-Security-398ca9a1453740068be5c7964a4059bb), the validator changes should be accumulated per consumer chain. 
-Like this, it would make it easier to have validators opting out from certain consumer chains. 
+- For each consumer chain, we store the consumer validator set that is currently (i.e., in this epoch) validating the 
+  consumer chain. For each validator in the set we store i) its voting power, and ii) the public key that it is 
+  using on the consumer chain during the current (i.e., ongoing) epoch.
```
Comment (p-offtermatt) [inline docs/docs/adrs/adr-014-epochs.md:34]:
```suggestion
  using on the consumer chain during the current (i.e., ongoing) epoch.
```
What happens when the validator changes the assigned key on a consumer during the epoch?

**Review by p-offtermatt — APPROVED** [review]
LGTM, one nit about a sentence that I find unclear.

**Review by sainoe — APPROVED** [review]
LGTM

File: docs/docs/adrs/adr-014-epochs.md (hunk 1 above)
Comment (insumity) [reply docs/docs/adrs/adr-014-epochs.md:34]: Good point. Added a sentence on this.

**Review by MSalopek — APPROVED** [review]
If you can, can you add some docs about to changing the params using a governance proposal?

**Comment by insumity** [comment]
> If you can, can you add some docs about to changing the params using a governance proposal?

@MSalopek Added a sentence ("This param can be changed through a _governance proposal_ to ...") on the fact that param can be changed and that it is bounded.

**Review by mpoke — APPROVED** [review]

**Pull request merged**